app/util/rpc-domain-utils.test.ts

```diff
@@ -1,152 +1,377 @@
     });
-
-    it('returns "unknown" when network client ID is not found', () => {
-      // Mock implementation for missing network client
-      (Engine.context.NetworkController.findNetworkClientIdByChainId as jest.Mock).mockReturnValue(null);
-
-      const result = getNetworkRpcUrl('0x999');
-
-      // Verify "unknown" is returned
-      expect(result).toBe('unknown');
-      expect(Engine.context.NetworkController.getNetworkConfigurationByNetworkClientId).not.toHaveBeenCalled();
+    it('initializes with empty set on error', async () => {
+      // Mock getSafeChainsListFromCacheOnly to throw
+      const spy = jest.spyOn({ getSafeChainsListFromCacheOnly }, 'getSafeChainsListFromCacheOnly').mockRejectedValueOnce(new Error('Test error'));
+      await initializeRpcProviderDomains();
+      expect(getKnownDomains()).toEqual(new Set());
+      spy.mockRestore();
     });
   });
-
-  describe('generateRPCProperties', () => {
-    it('returns { properties: { rpc_domain: "example.com" }, sensitiveProperties: {} } for a known domain', () => {
-      jest.spyOn(rpcUtils, 'getNetworkRpcUrl').mockReturnValue('https://example.com');
-      jest.spyOn(rpcUtils, 'extractRpcDomain').mockReturnValue('example.com');
-      expect(generateRPCProperties('0x1')).toEqual({
-        properties: { rpc_domain: 'example.com' },
-        sensitiveProperties: {},
+  describe('getKnownDomains and setKnownDomains', () => {
+    describe('when setting known domains', () => {
+      // Removed tests that mock getKnownDomains directly as they are not robust or meaningful
+    });
+  });
+  describe('isKnownDomain', () => {
+    describe('when checking domain existence', () => {
+      beforeEach(async () => {
+        resetModuleState();
+        const mockChains: SafeChain[] = [
+          {
+            chainId: '1',
+            name: 'Test Chain',
+            nativeCurrency: { symbol: 'TEST' },
+            rpc: ['https://known-domain.com/api'],
+          },
+        ];
+        (StorageWrapper.getItem as jest.Mock).mockResolvedValue(JSON.stringify(mockChains));
+        await initializeRpcProviderDomains();
+      });
+      it('returns true for known domains', () => {
+        // Execute
+        const result = isKnownDomain('known-domain.com');
+        // Verify
+        expect(result).toBe(true);
+      });
+      it('returns false for unknown domains', () => {
+        // Execute
+        const result = isKnownDomain('unknown-domain.com');
+        // Verify
+        expect(result).toBe(false);
+      });
+      it('handles null knownDomainsSet', async () => {
+        // Setup
+        resetModuleState();
+        (StorageWrapper.getItem as jest.Mock).mockResolvedValue(null);
+        await initializeRpcProviderDomains();
+        // Execute
+        const result = isKnownDomain('any-domain.com');
+        // Verify
+        expect(result).toBe(false);
+      });
+      it('performs case-insensitive domain matching', async () => {
```

Comment:
S: don't hesitate to simplify, the significant action verb at the begining. "perform" add no meaning here.

```suggestion
      it('matches domain ignoring case', async () => {
```